docs/developer/architecture/snmp.md

```diff
@@ -1,3 +1,102 @@
 # SNMP
 
------
+!!! note
+    This section is meant for developers that want to understand the working of the SNMP integration.
+
+    Be sure you are familiar with [SNMP concepts](../../tutorials/snmp/introduction/), and you have read through the official [SNMP integration docs](https://docs.datadoghq.com/integrations/snmp).
+
+## Overview
```

Comment:
Seems we should also mentioned the profiles or ["metrics" configuration](https://github.com/DataDog/integrations-core/blob/fd6df90135de14d06662e49d7696a42c08476a82/snmp/datadog_checks/snmp/data/conf.yaml.example#L344-L354). 

And point to existing docs about how to build profiles.